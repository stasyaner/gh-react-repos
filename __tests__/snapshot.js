// DISCLAIMER: it's a JS because it's hard to setup TS for snapshots
// and doesn't really make sense here

import {
    act,
    create,
} from "react-test-renderer";
import fetch from "jest-fetch-mock";
import Home from "../pages";
import { testReactReturn } from "./__test_data__";

jest.mock("next/router", () => ({
    useRouter() {
        return {
            query: "",
        };
    },
}));

describe("snapshot testing", () => {
    it("renders home page", async () => {
        fetch.mockResponseOnce(JSON.stringify(testReactReturn));

        let tree;
        await act(async () => {
            tree = create(<Home />);
        });

        expect(tree.toJSON()).toMatchSnapshot();

        jest.clearAllMocks();
        fetch.mockClear();
    });
});
