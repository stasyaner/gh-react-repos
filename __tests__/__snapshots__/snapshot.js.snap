// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing renders home page 1`] = `
<main
  className="container my-5"
>
  <ul>
    <li>
      <a
        href="https://github.com/facebook/react"
      >
        react
      </a>
       - 
       
      <span
        aria-label="star"
        role="img"
      >
        🌟
      </span>
       
      162004
       - 
       
      <span
        aria-label="fork"
        role="img"
      >
        🍴
      </span>
       
      32317
    </li>
  </ul>
  <nav
    aria-label="repo list navigation"
    className="d-flex justify-content-center"
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item disabled"
      >
        <span
          className="page-link"
        >
          <span
            aria-hidden="true"
          >
            «
          </span>
          <span
            className="sr-only"
          >
            Previous
          </span>
        </span>
      </li>
      <li
        className="page-item active"
      >
        <a
          className="page-link"
          href="/?page=1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          1
        </a>
      </li>
      <li
        className="page-item disabled"
      >
        <span
          className="page-link"
        >
          <span
            aria-hidden="true"
          >
            »
          </span>
          <span
            className="sr-only"
          >
            Next
          </span>
        </span>
      </li>
    </ul>
  </nav>
</main>
`;
